config/reload.go

```diff
@@ -49,10 +50,15 @@ func GenerateChecksum(yamlFilePath string) (string, error) {
 	dir := filepath.Dir(yamlFilePath)
 
 	for i, file := range config.RuleFiles {
-		config.RuleFiles[i] = filepath.Join(dir, file)
+		if !strings.Contains(config.RuleFiles[i], dir) {
```

Comment:
why `config.RuleFiles[i]` and not `file`?

Reply:
I actually am not sure what the exact value of `file` is but I figured it would get us close enough to detect changes if we used the value supplied in the config.

Reply:
You got me thinking, and I'll try it out to see how it works because now I'm concerned that it would reload for every file that had changes (which I didn't test). I really hope it doesn't do that heh

Reply:
As for `config.JoinDir()`, I had no idea that was there, so thanks! My modifications were only what we had there. Curious to know what the difference/benefit would be of changing this code to use `config.JoinDir()` instead.